[hardknott] merge audit syntax and styling commits

amstewart opened the pull request:
This PR contains all of the "syntax and styling" commits from the meta-nilrt layer audit, as well as the commits which rename recipe files from `$name_1.0bb` to `$name.bb`. Though it contains many commits, none of them should result in substantial behavioral changes to the feed IPKs or images, with one exception.

While re-styling the `salt` recipe, I noticed that the `INITSCRIPT` settings for the salt minion were conflated when we moved the recipe from meta-openstack to meta-nilrt. The minion [should have been stopping](https://github.com/ni/meta-nilrt/commit/92e0813a2d1a708a3838e5bf8edb49f0ffc11c03) at priority `93`, but was instead using the `default` stop priority. I fixed that during the styling and confirmed in the new runmode image that the priority is back to 93.

NI AZDO: https://dev.azure.com/ni/DevCentral/_workitems/edit/2116516

# Testing
* Performed a local build of the core feed and images. Manually compared the feed indexes and image manifests for differences. The only observed differences (apart from the salt change above) were expected fixups to package descriptions and information metadata (eg. `Homepage`).

Commit:
base-files: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
busybox: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
coreutils: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
distro-feed-configs; syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
glibc: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

amstewart committed:
apr-util: variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
apr-iconv: variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ntp: variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
onboard: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
salt: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
strongswan: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
syslog-ng: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
trousers: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
base-files-nilrt: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
crio-support-scripts: rename recipe

There is no reason to encode the package version into the recipe file
name for crio-support-scripts. Rename it to remove the encoding.

Also rename the source files directory to match the recipe name, to
support the recipe's eventual move into the ni-hw-scripts family of
recipes.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
crio-support-scripts: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
crio-support-scripts: add PV variable

Explicitly set the package version via a variable within the recipe, so
that devs can arbitrarily rev it when they make changes.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
initscripts-nilrt: remove version from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
initscripts-nilrt-safemode: remove version from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
initscripts-nilrt: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
initscripts-nilrt-safemode: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-grpc-device: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-hw-scripts-common: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-hw-scripts-common: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-hw-scripts: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-hw-scripts: variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-modules-autoload: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-modules-autoload: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-netcfgutil: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-netcfgutil: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
nirtcfg-tests: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
nirtcfg-tests: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-safemode-utils: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-safemode-utils: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-shutdown-guard: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-shutdown-guard: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-smbios-helper: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemformat: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemformat: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemimage: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemimage: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
nisystemformat: fixup whitespace

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
fw-printenv: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
fw_printenv: fixup whitespace

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-utils: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

ckoellin committed:
xfce: recipe styling and syntax

Signed-off-by: chris.koelling <chris.koelling@ni.com>

amstewart committed:
ni-utils: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
systemimageupdateinfo: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
systemimageupdateinfo: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niwatchdogpet: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niwatchdogset: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pstore-save: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pstore-save: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
sysconfig-settings: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
sysconfig-settings: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
transconf: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
transconf: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

Commit:
init-ifupdown: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
initrdscripts: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

ckoellin committed:
graphics: recipe styling and syntax

Signed-off-by: chris.koelling <chris.koelling@ni.com>

Commit:
meta: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
rauc: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
readline: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
sysvinit: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

A variable was moved within the recipe upstream; trivially relocated the
diff.

Rebased-by: Alex Stewart <alex.stewart@ni.com>

Commit:
udev: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

Commit:
util-linux: syntax and styling

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

amstewart committed:
nativesdk-packagegroup-sdk-host: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
packagefeed-ni-*: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

ckoellin committed:
rt: recipe styling and syntax

Signed-off-by: chris.koelling <chris.koelling@ni.com>

amstewart committed:
packagegroups: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

Commit:
images: styling and syntax

Signed-off-by: Charlie Johnston <charlie.johnston@ni.com>

ckoellin committed:
kernel: recipe styling and syntax

Signed-off-by: chris.koelling <chris.koelling@ni.com>

Upstream already slightly restyled the dkms recipe. Trivially reconciled
the two styling changes.

Rebased-by: Alex Stewart <alex.stewart@ni.com>

mike-petersen-ni commented on recipes-core/busybox/busybox_1.33.%.bbappend:
```diff
@@ -1,50 +1,51 @@
-            file://zcip-allow-action-script-to-reject-chosen-IP.patch \
-            file://login-utilities.cfg \
-            file://udhcpd.wlan0.conf"
+			file://busybox-ifplugd \
```
Are triple-tabs intentional?
Multi-tabs also appear in:

    recipes-core/images/includes/nilrt-proprietary.inc
    recipes-core/util-linux/util-linux_2.%.bbappend

amstewart replied:
I think @SparkingSpork did the `recipes-core/` and chose to do it that way. I can convert them to single-tabs.

mike-petersen-ni commented on recipes-ni/crio-support-scripts/crio-support-scripts.bb:
```diff
@@ -0,0 +1,33 @@
+do_install () {
+	install -d ${D}${sysconfdir}/init.d/
+	if [ "${TARGET_ARCH}" = "x86_64" ]; then
+		install -m 0755   ${S}/nisetfpgaautoload	${D}${sysconfdir}/init.d
+		install -m 0550   ${S}/nisetconsoleout	  ${D}${sysconfdir}/init.d
```
Not entirely relevant to this PR, but why the difference in permissions?

amstewart replied:
Just a historic quirk, I think.

mike-petersen-ni commented on recipes-ni/crio-support-scripts/crio-support-scripts.bb:
```diff
@@ -0,0 +1,33 @@
+PV = "2.0"
+
+
+SRC_URI = "\
```
Looks like we lost an `_append_x64` -> `:append:x64` here. The commit 768e4351ef1bd0cef8770fb00a6c4f51009ebd62 did not mention it was intentional; looks like a possible mistake.

amstewart replied:
Since we only support x64 in the hardknott mainline, it was an unnecessary complication. I'm going to retain this change, unless you have an objection.

mike-petersen-ni commented on recipes-ni/ni-systemimage/ni-systemimage.bb:
```diff
@@ -1,17 +1,22 @@
-SUMMARY = "A system imaging utility for NI LinuxRT safemodes"
+SUMMARY = "A system imaging utility for NI NILRT safemodes"
```
"NI NILRT" seems a bit redundant.

amstewart replied:
Will fix.

mike-petersen-ni commented on recipes-ni/ni-systemimage/ni-systemimage.bb:
```diff
@@ -23,9 +28,12 @@ do_install () {
+RDEPENDS:${PN} += "\
+	bash \
+	ni-netcfgutil \
+	niacctbase \
```
Added dependency; why?

amstewart replied:
The `do_install()` uses the `${LVRT_GROUP}` variable - which is only meaningful when the `niacctbase` package is installed. `niacctbase` sets up the LVRT user, group, and home directory.

shruthi-ravi replied:
It'd be nice if you could call that out in the commit description and explain why.

mike-petersen-ni commented on recipes-ni/ni-utils/fw-printenv_1.0.bb:
```diff
@@ -1,37 +1,41 @@
-
-FILES_${PN} += "${datadir}/fw_printenv/*"
-
-DEPENDS += "shadow-native pseudo-native niacctbase"
```
Looks like we lost a couple dependencies here.

amstewart replied:
`shadow-native` and `pseudo-native` aren't needed for recipes which only install content under uid 0 - like this one. The deps were added to this recipe - and many others - in a bulk commit from years ago, which wasn't conceptually correct.

mike-petersen-ni commented on recipes-ni/nirtcfg-tests/nirtcfg-tests_1.0.bb:
```diff
@@ -1,45 +0,0 @@
-# We only want to build the -ptest package
-PACKAGES_remove = "${PN}-dev ${PN}-staticdev ${PN}-dbg"
-
-RDEPENDS_${PN}-ptest_append = " bash"
```
Looks like a lost `_append` suffix.

amstewart replied:
`VARIABLE += VALUE` is logically equivalent to `VARIABLE:append = value` in simple cases like this (ie. there are subtle differences, but they don't apply here). When doing simple operations like this, using the `+=` operator is generally preferred.

mike-petersen-ni commented on recipes-ni/nirtcfg-tests/nirtcfg-tests_1.0.bb:
```diff
@@ -1,45 +0,0 @@
-LICENSE = "MIT"
-LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
-
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Lost this line.

amstewart replied:
`./files` and `./${PN}` are in the `FILES` path by default for recipes. Whoever authored this recipe added them to `FILESEXTRAPATHS`, because that is what you need to do for **bbappend** files, but not for `.bb`s. It isn't needed.

mike-petersen-ni commented on recipes-ni/pstore-save/pstore-save.bb:
```diff
@@ -0,0 +1,47 @@
+
+
+do_install () {
+	install -d ${D}${sbindir}
```
Appears to be redundant.

amstewart replied:
Will fix.

mike-petersen-ni reviewed (commented):
Haven't looked at salt changes yet.

If we ever do this again in the future, perhaps we should consider splitting into several stages to keep clean diffs:
 - indenting and newlining
 - combining appends
 - reordering
 - semantic changes

amstewart committed:
busybox: use single-tabs for indentation

Whitespace changes only.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemimage: fixup redundant SUMMARY text

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pstore-save: remove redundant install operation

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V2
* Used single tabs in the `busybox` bbappend. @mike-petersen-ni 
* Removed redundant SUMMARY text in `ni-systemimage`. @mike-petersen-ni 
* Removed redundant `install` operation in `pstore-save`. @mike-petersen-ni

shruthi-ravi commented on recipes-kernel/packagegroups/packagegroup-ni-debug-kernel.bb:
```diff
@@ -1,16 +1,18 @@
 "
 
-RDEPENDS_${PN}_append_armv7a = " \
+RDEPENDS:${PN}:append_armv7a = " \
```
Should this be `append:armv7a`?

amstewart replied:
Good spot; will fix.

amstewart replied:
Should be fixed in #467

shruthi-ravi commented on recipes-ni/ni-safemode-utils/ni-safemode-utils.bb:
```diff
@@ -1,33 +1,39 @@
 LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
-SECTION = "base"
+
+DEPENDS = "niacctbase"
```
Why have the remaining packages been removed from this list?

amstewart replied:
`shadow-native` and `pseudo-native` were bulk-added to a bunch of recipes many many years ago to fix some issue with how `pseudo` behaved. Since then, the issue has been resolved upstream and these `DEPENDS` are no longer needed. The `update-rc.d-native` DEPENDS is not needed because this recipe doesn't call `update-rc.d` anywhere.

I'll add those justifications to the commit message.

shruthi-ravi commented on recipes-ni/ni-systemformat/ni-systemformat.bb:
```diff
@@ -25,21 +34,18 @@ do_install () {
-RDEPENDS_${PN} += "bash"
+RDEPENDS:${PN} += " \
+	bash \
+	niacctbase \
```
Why have you moved this from the DEPENDS list? Your commit description does not say why.

shruthi-ravi reviewed (approved):
I'm still not convinced that most of these changes (except the syntax changes mandated by upstream) were worth all of our time added up, especially the newline and spacing updates. I disagree with your tab spacing updates for reasons I mentioned in your style documentation PR. I would be less resistant if you were making the whitespace changes to maintain consistency within a file, but to touch nearly every file in this repo and enforce that is just not sustainable.

I hope next time you are able to categorize your changes into different sets beforehand (like mandated syntax updates, whitespace changes, reorg changes, etc.). In some commits you've seem to have made more than just syntax and styling updates.

mike-petersen-ni reviewed (approved):
I'll take your word for it on `recipes-support/salt/salt_3000.2.bb`; log+diff are too mangled.

amstewart commented:
I've decided to abandon this approach to merging the layer audit in favor of a more piecemeal approach. See the later `audit 2*` PRs, beginning with #467 .
